include/tvm/node/structural_equal.h

```diff
@@ -56,6 +57,31 @@ class BaseValueEqual {
   }
 };
 
+/*!
+ * \brief Pair of `ObjectPath`s, one for each object being tested for structural equality.
+ */
+struct ObjectPathPair {
+  ObjectPath lhs_path;
+  ObjectPath rhs_path;
+};
+
+// Could be replaced with std::optional<ObjectPathPair>
+class OptionalObjectPathPair {
+ public:
+  OptionalObjectPathPair() = default;
+
+  OptionalObjectPathPair(const ObjectPathPair& p)  // NOLINT(runtime/explicit)
+      : lhs_path(p.lhs_path), rhs_path(p.rhs_path) {}
+
+  bool defined() const { return lhs_path.defined(); }
+
+  ObjectPathPair value() const { return {lhs_path.value(), rhs_path.value()}; }
+
+ private:
+  Optional<ObjectPath> lhs_path;
+  Optional<ObjectPath> rhs_path;
+};
+
```

Comment:
Quick question: if `ObjectPathPair` is a TVM object, then we don't need to define another class as the optional wrapper - is my understanding correct?

Reply:
That's correct, but making it a TVM object would require heap allocation.